### ISSUE TITLE
I have some vimeo.com/showcase-Links, that didn't work

### ISSUE THREAD

**Issue by kiri35** [LeoniePhiline/showcase-dl#11]
I have some vimeo.com/showcase-Links, that didn't work. Plrease give me an email-adress to send they to you.

**Comment by LeoniePhiline** [comment]
Try `showcasedl-11@nervmich.net`

**Comment by kiri35** [comment]
(I have translated the following text with deepl.com. The original in German is below).
 

Hello,

0) thanks a lot for your program showcase-dl :-) The original error is not a bug, but my misoperation. I have described it under 2). Additionally I have suggestions/wishes for 4 switches (options):
 

1) I really like that I can run multiple yt-dlp calls in parallel with it. Maybe the following switch would be helpful?:
-mt count, --maxThreads count.

And I would find it helpful if the "Finished" videos (shown in green) would scroll out the top of the window, so I can see the current ones at the bottom if the terminal window is too small for many threads running in parallel.

Ps: instead of "Finished" it currently shows "Finishe".
 

2) This page (if the link is down and you need a new one, please write me)

https://shop.auditorium-netzwerk.de/custom/index/sCustom/992
contains these 3 video links
https://vimeo.com/showcase/9643839/embed
https://player.vimeo.com/video/454333258
https://player.vimeo.com/video/722187875?h=bcde0752cf

If I call them like this
~/.local/bin/showcase-dl/target/release/showcase-dl https://vimeo.com/showcase/9643839/embed
~/.local/bin/showcase-dl/target/release/showcase-dl https://player.vimeo.com/video/454333258
then showcase-dl does not work and I do not see any error message or reference to the following:

It took me some time to discover that showcase-dl automatically :-) parses the page and downloads all viedos with
~/.local/bin/showcase-dl/target/release/showcase-dl https://shop.auditorium-netzwerk.de/custom/index/sCustom/992

E.g. such a hint in the showcase-dl window - if a call "showcase-dl video link" does nothing - would have helped me to detect it earlier: 
"showcase-dl WebPage" analyzes the whole web page and downloads all videos from it.
 

3) I need 3 switches (options) in showcase-dl:

--notExtractAudio, or better that as default and then: -ea, --extractAudio.
(maybe we can pass the two bitrates for .opus and for .mp3)

Passing switches (options) to yt-dlp for -f, --format and for --audio-multistreams to be able to pass the desired format to yt-dlp, so I could do e.g. the following calls:
showcase-dl -f bv[vcodec^=avc1]+ba[acodec^=mp4a]/b https://shop.auditorium-netzwerk.de/custom/index/sCustom/992
showcase-dl --audio-multistreams -f bv[vcodec^=avc1]+ba[acodec^=opus]+ba[acodec^=mp4a]/b https://shop.auditorium-netzwerk.de/custom/index/sCustom/992

And the switch from above from 1) would be helpful for me:
showcase-dl -mt 6 -f bv[vcodec^=avc1]+ba[acodec^=mp4a]/b https://shop.auditorium-netzwerk.de/custom/index/sCustom/992
 

4) Nice to have, but not essential: I would like to have 1 switch (option) -w file.txt, --write file.txt in showcase-dl for outputting the found video links:

showcase-dl --write File.txt https://shop.auditorium-netzwerk.de/custom/index/sCustom/992
writes the following text file:

#!/bin/bash
yt-dlp ... 2 lines for all found videos from https://vimeo.com/showcase/9643839/embed ...
yt-dlp https://player.vimeo.com/video/454333258
yt-dlp https://player.vimeo.com/video/722187875?h=bcde0752cf

and
showcase-dl -w file.txt --maxThreads 3 --audio-multistreams -f bv[vcodec^=avc1]+ba[acodec^=opus]+ba[acodec^=mp4a]/b https://shop.auditorium-netzwerk.de/custom/index/sCustom/992
writes the following text file:

#!/bin/bash
yt-dlp --audio-multistreams -f bv[vcodec^=avc1]+ba[acodec^=opus]+ba[acodec^=mp4a]/b ... 2 lines for all found videos from https://vimeo.com/showcase/9643839/embed ...
yt-dlp --audio-multistreams -f bv[vcodec^=avc1]+ba[acodec^=opus]+ba[acodec^=mp4a]/b https://player.vimeo.com/video/454333258
#yt-dlp --audio-multistreams -f bv[vcodec^=avc1]+ba[acodec^=opus]+ba[acodec^=mp4a]/b https://player.vimeo.com/video/722187875?h=bcde0752cf
 

Thanks again and have a nice day, Kiri

 

--------------------------------------------------------------------------------

(Ich habe den folgenden Text mit deepl.com übersetzt. Das Original in Deutsch steht unten.)

Hallo,

0) vielen Dank für ihr Programm showcase-dl :-) Der ursprüngliche Fehler ist kein Fehler, sondern meine Fehlbedienung. Ich habe ihn unter 2) beschrieben. Zusätzlich habe ich Vorschläge/Wünsche für 4 Schalter (Optionen):
 

1) Es gefällt mir sehr gut, dass ich damit mehrere yt-dlp Aufrufe prallel laufen lassen kann. Vielleicht wäre der folgende Schalter hilfreich?:
-mt Anzahl, --maxThreads Anzahl

Und ich fände es hilfreich, wenn die "Finished" Videos (in grün angezeigt) nach oben aus dem Fenster rausscrollen würden, sodass ich die aktuellen unten sehen kann, wenn das Terminal-Fenster zu klein ist für viele parallel laufende Threads.

Ps: statt "Finished" erscheint zur Zeit "Finishe".
 

2) Diese Seite (falls der Link down ist und du einen neuen brauchst, dann schreibe mir bitte)
https://shop.auditorium-netzwerk.de/custom/index/sCustom/992

enthält diese 3 Video-Links
https://vimeo.com/showcase/9643839/embed
https://player.vimeo.com/video/454333258
https://player.vimeo.com/video/722187875?h=bcde0752cf

Wenn ich sie so aufrufe
~/.local/bin/showcase-dl/target/release/showcase-dl https://vimeo.com/showcase/9643839/embed
~/.local/bin/showcase-dl/target/release/showcase-dl https://player.vimeo.com/video/454333258
~/.local/bin/showcase-dl/target/release/showcase-dl https://player.vimeo.com/video/722187875?h=bcde0752cf
dann arbeitet showcase-dl nicht und ich sehe keine Fehlermeldung oder einen Hinweis auf das folgende:

Ich habe etwas Zeit gebraucht, bis ich entdeckt habe, das showcase-dl automatisch :-) die Seite analysiert und alle Viedos runterlädt mit
~/.local/bin/showcase-dl/target/release/showcase-dl https://shop.auditorium-netzwerk.de/custom/index/sCustom/992

Z.B. so ein Hinweis im showcase-dl Fenster – wenn ein Aufruf "showcase-dl video link" nichts tut – hätte mir geholfen, das früher zu entdecken:
"showcase-dl WebPage" analysiert die ganze Web-Seite und läd alle Videos runter.
 

3) Ich brauche 3 Schalter (Optionen) in showcase-dl:

--notExtractAudio, oder besser das als default und dann: -ea, --extractAudio
(vielleicht können wir die beiden Bitraten übergeben für .opus und für .mp3)

Das Weiterreichen von Schaltern (Optionen) an yt-dlp für -f, --format und für --audio-multistreams, um das gewünschte Format an yt-dlp weiter geben zu können, sodass ich z.B. folgende Aufrufe machen könnte:
showcase-dl -f bv[vcodec^=avc1]+ba[acodec^=mp4a]/b https://shop.auditorium-netzwerk.de/custom/index/sCustom/992
showcase-dl --audio-multistreams -f bv[vcodec^=avc1]+ba[acodec^=opus]+ba[acodec^=mp4a]/b https://shop.auditorium-netzwerk.de/custom/index/sCustom/992

Und der Schalter von oben aus 1) wäre hilfreich für mich:
showcase-dl -mt 6 -f bv[vcodec^=avc1]+ba[acodec^=mp4a]/b https://shop.auditorium-netzwerk.de/custom/index/sCustom/992
 

4) Schön zu haben, aber nicht unbedingt notwendig: Ich hätte gerne 1 Schalter (Option) -w Datei.txt, --write Datei.txt in showcase-dl für die Ausgabe der gefundenen Video-Links:

showcase-dl --write Datei.txt https://shop.auditorium-netzwerk.de/custom/index/sCustom/992
schreibt folgende Textdatei:

#!/bin/bash
yt-dlp … 2 Zeilen für alle gefundenen Videos aus https://vimeo.com/showcase/9643839/embed …
yt-dlp https://player.vimeo.com/video/454333258
yt-dlp https://player.vimeo.com/video/722187875?h=bcde0752cf

und
showcase-dl -w Datei.txt --maxThreads 3 --audio-multistreams -f bv[vcodec^=avc1]+ba[acodec^=opus]+ba[acodec^=mp4a]/b https://shop.auditorium-netzwerk.de/custom/index/sCustom/992
schreibt folgende Textdatei:

#!/bin/bash
yt-dlp --audio-multistreams -f bv[vcodec^=avc1]+ba[acodec^=opus]+ba[acodec^=mp4a]/b … 2 Zeilen für alle gefundenen Videos aus https://vimeo.com/showcase/9643839/embed …
yt-dlp --audio-multistreams -f bv[vcodec^=avc1]+ba[acodec^=opus]+ba[acodec^=mp4a]/b https://player.vimeo.com/video/454333258
#yt-dlp --audio-multistreams -f bv[vcodec^=avc1]+ba[acodec^=opus]+ba[acodec^=mp4a]/b https://player.vimeo.com/video/722187875?h=bcde0752cf

Vielen Dank nochnmal und einen schönen Tag, Kiri





Gesendet: Montag, 26. Juni 2023 um 14:13 Uhr
Von: "LeoniePhiline" ***@***.***>
An: "LeoniePhiline/showcase-dl" ***@***.***>
Cc: "kiri35" ***@***.***>, "Author" ***@***.***>
Betreff: Re: [LeoniePhiline/showcase-dl] I have some vimeo.com/showcase-Links, that didn't work (Issue #11)


 

Try ***@***.***

**Comment by kiri35** [comment]
This email-adress doesn't work:

***@***.***

 

Now the links are public.

 
 

Gesendet: Montag, 26. Juni 2023 um 14:13 Uhr
Von: "LeoniePhiline" ***@***.***>
An: "LeoniePhiline/showcase-dl" ***@***.***>
Cc: "kiri35" ***@***.***>, "Author" ***@***.***>
Betreff: Re: [LeoniePhiline/showcase-dl] I have some vimeo.com/showcase-Links, that didn't work (Issue #11)


 

Try ***@***.***

**Comment by LeoniePhiline** [comment]
Instead of `--write FILE` you can just use one or more `-v` for log verbosity and grep/filter the log for the spawned subprocess commands.

**Comment by LeoniePhiline** [comment]
# Status: 

- [ ] https://github.com/LeoniePhiline/showcase-dl/issues/12
- [x] https://github.com/LeoniePhiline/showcase-dl/issues/13
- [x] https://github.com/LeoniePhiline/showcase-dl/issues/14
- [x] https://github.com/LeoniePhiline/showcase-dl/issues/15
- [ ] https://github.com/LeoniePhiline/showcase-dl/issues/16
